wandb/sync/sync.py

```diff
@@ -70,20 +77,118 @@ def __init__(
         self._verbose = verbose
         self._mark_synced = mark_synced
         self._app_url = app_url
+        self._sync_tensorboard = sync_tensorboard
+
+    def _parse_pb(self, data, exit_pb=None):
+        pb = wandb_internal_pb2.Record()
+        pb.ParseFromString(data)
+        record_type = pb.WhichOneof("record_type")
+        if self._view:
+            if self._verbose:
+                print("Record:", pb)
+            else:
+                print("Record:", record_type)
+            return pb, exit_pb, True
+        if record_type == "run":
+            if self._run_id:
+                pb.run.run_id = self._run_id
+            if self._project:
+                pb.run.project = self._project
+            if self._entity:
+                pb.run.entity = self._entity
+            pb.control.req_resp = True
+        elif record_type == "exit":
+            exit_pb = pb
+            return pb, exit_pb, True
+        elif record_type == "final":
+            assert exit_pb, "final seen without exit"
+            pb = exit_pb
+            exit_pb = None
+        return pb, exit_pb, False
+
+    def _find_tfevent_files(self, sync_item):
+        tb_event_files = 0
+        tb_logdirs = []
+        tb_root = None
+        if self._sync_tensorboard:
+            # TODO: complain if there are too many dirs...
+            if os.path.isdir(sync_item):
+                glob_str = os.path.join(
+                    sync_item, "**", "*{}*".format(TFEVENT_SUBSTRING)
+                )
+                files = glob.glob(glob_str, recursive=True)
+                for tfevent in files:
+                    tb_event_files += 1
+                    tb_dir = os.path.dirname(os.path.abspath(tfevent))
+                    if tb_dir not in tb_logdirs:
+                        tb_logdirs.append(tb_dir)
+                if len(tb_logdirs) > 0:
+                    tb_root = os.path.commonprefix(tb_logdirs)
+            elif TFEVENT_SUBSTRING in sync_item:
+                tb_root = os.path.dirname(os.path.abspath(sync_item))
+                tb_logdirs.append(tb_root)
+                tb_event_files = 1
+        return tb_event_files, tb_logdirs, tb_root
+
+    def _send_tensorboard(self, tb_root, tb_logdirs, sm, publish_interface, record_q):
```

Comment:
Is there any reason why the record_q should be different than the record queue of the publish_interface? If not, consider removing this param and just getting the record_q off the publish_interface object.